public_html/wp-content/plugins/wordcamp-payments-network/includes/sponsor-invoices-dashboard.php

```diff
@@ -461,3 +464,107 @@ function delete_index_row( $invoice_id ) {
 		)
 	);
 }
+
+/**
+ * Schedule cron to send reminder mails to organizers for unpaid invoices.
+ */
+function schedule_sent_invoice_reminder() {
+	if ( wp_next_scheduled( 'send_invoice_pending_reminder' ) ) {
+		return;
+	}
+
+	wp_schedule_event( time(), 'daily', 'send_invoice_pending_reminder' );
+}
+
+/**
+ * Send reminder to organizer about the unpaid invoice.
+ */
+function send_invoice_pending_reminder() {
+	global $wpdb;
+
+	$table_name    = get_index_table_name();
+	$sent_invoices = $wpdb->get_results( "
+		SELECT blog_id, invoice_id
+		FROM $table_name
+		WHERE status = 'wcbsi_approved'
+		LIMIT 1000
+	" );
+
+	foreach ( $sent_invoices as $invoice_data ) {
+		$blog_id    = $invoice_data->blog_id;
+		$invoice_id = $invoice_data->invoice_id;
+
+		switch_to_blog( $blog_id );
+
+		$invoice         = get_post( $invoice_id );
+		$invoice_sent_at = get_post_meta( $invoice_id, 'Sent at', true );
+
+		if ( empty( $invoice_sent_at ) ) {
+			// Backfill for older invoices.
+			update_post_meta( $invoice_id, 'Sent at', time() );
+			update_post_meta( $invoice_id, 'Backfilled Sent at', true );
+			restore_current_blog();
+			continue;
+		}
+
+		$last_reminder  = get_post_meta( $invoice_id, 'last_reminder_details', true );
+		$reminder_step  = 1;
+		$last_step_time = $invoice_sent_at;
+		if ( ! empty( $last_reminder ) ) {
+			$reminder_step  = $last_reminder['step'] + 1;
+			$last_step_time = $last_step_time['sent_at'];
+		}
+
+		// We will send reminders after 30, 45, and 60 days.
+		$reminder_schedule = array(
+			1 => 30,
+			2 => 15,
+			3 => 15,
+		);
+
+		if ( $reminder_step > count( $reminder_schedule ) ) {
+			// At this point 2 months have been passed. Maybe send a mail to Community support at this point.
```

Comment:
It sounds like Andrea wants to do this. This function is already pretty long, so maybe at this point would should modularize it a bit. Maybe in this condition you could call a new function that handles everything related to sending the email to central. Then you could create another function that contains all the code below that's related to sending the email to the organizer?

Reply:
thanks for the suggestion. done.